File: v2/src/main/java/com/api/v2/people/domain/PersonRepository.java
package com.api.v2.people.domain;

import com.api.v2.people.domain.exposed.Person;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

public interface PersonRepository extends ReactiveMongoRepository<Person, String> {

    @Query("{ 'ssn': ?0, 'email': ?1 }")
    Mono<Person> find(String ssn, String email);

}

File: v2/src/main/kotlin/com/api/v2/people/domain/PersonRepository.kt
package com.api.v2.people.domain

import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.kotlin.CoroutineCrudRepository

interface PersonRepository: CoroutineCrudRepository<Person, String> {

    @Query("{ '\$or': [ { 'sin': ?0, 'email': ?1 } ] }")
    suspend fun findBySINOrEmail(sin: String, email: String): Person?
}